rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -42,6 +42,8 @@ http {
     lua_shared_dict configuration_data 5M;
     lua_shared_dict round_robin_state 1M;
     lua_shared_dict locks 512k;
+    lua_shared_dict balancer_ewma 1M;
+    lua_shared_dict balancer_ewma_last_touched_at 1M;
```

Comment:
Is this size enough? Do you have an estimation of how many endpoints fit in 1M?

Reply:
@shivnagarajan Do you know the answer to any of these questions?

Reply:
@zrdaley you can do some back of the napkin math here. `balancer_ewma` holds key value data where keys are upstream names and values are EWMAs(float value). Similarly for `balancer_ewma_last_touched_at` where keys are the same and values are time.

> What happens if the dict is full?

When the dict is full we fail to store the EWMA values which means the load balancing will start misbehaving.

Reply:
`balancer_ewma` and `balancer_ewma_last_touched_at` are both dictionaries that have a string key and floating-point value. 

>Do you have an estimation of how many endpoints fit in 1M?

Each dict entry
~10 char key + float = 4B*10 + 4B
= 44B = 0.000044MB

Total entries
1MB/0.000044MB = **22,727**

Reply:
> What happens if the dict is full?

When the dict is full, values will be overwritten **or** fail to write entirely. I created nginx logging around `round_robin_state`, `balancer_ewma` and `balancer_ewma_last_touched_at` for both of these cases (see [round_robin_state example](https://github.com/kubernetes/ingress-nginx/pull/2229/files#L63-L69)).

The full `ngx.shared.DICT.set` docs can be found [here](https://github.com/openresty/lua-nginx-module#ngxshareddictset).